lib/spark.js

```diff
@@ -51,6 +51,10 @@ export default class Spark {
           stats.byteLength += value.byteLength
           resetTimeout()
         }
+      } else {
+        console.error({
+          error: await res.text()
+        })
```

Comment:
Interesting, I haven't seen this pattern before. IIUC, this will inspect the object and print its properties. The downside is that the error text may be printed with escaped `"` character.

```
❯ deno
Deno 1.32.0
> console.error({error: 'foo"bar'})
{ error: 'foo"bar' }
undefined
> console.error({error: 'foo"bar\''})
{ error: `foo"bar'` }
undefined
> console.error({error: 'foo"bar\'`'})
{ error: "foo\"bar'`" }
undefined
>
```

Have you considered printing a descriptive message instead?

```suggestion
        console.error('Retrieval failed with status code %s: %s', 
          res.status, await res.text())
```

Reply:
No particular reason to do it as I did here. There was no problem with escaping either. Your solution has better DX though